src/borg/archiver.py

```diff
@@ -1016,18 +1016,26 @@ def do_prune(self, args, repository, manifest, key):
         stats = Statistics()
         with Cache(repository, key, manifest, do_files=args.cache_files, lock_wait=self.lock_wait) as cache:
             list_logger = logging.getLogger('borg.output.list')
+            archives_to_delete = []
             for archive in archives_checkpoints:
                 if archive in to_delete:
                     if args.dry_run:
                         if args.output_list:
                             list_logger.info('Would prune:     %s' % format_archive(archive))
                     else:
-                        if args.output_list:
-                            list_logger.info('Pruning archive: %s' % format_archive(archive))
-                        Archive(repository, key, manifest, archive.name, cache).delete(stats, forced=args.forced)
+                        archives_to_delete.append(archive)
                 else:
                     if args.output_list:
                         list_logger.info('Keeping archive: %s' % format_archive(archive))
+            if args.progress:
+                len_archives_to_delete = len(archives_to_delete)
```

Comment:
isn't this the same as len(to_delete)?

if so, can you stay close to the old code to do what you want?